### ISSUE TITLE
[Bug]: Pressing Enter Key consecutively 3-4 times will change picker value to label, instead of the default one.

### ISSUE THREAD

**Issue by neetikajain27** [adobe/spectrum-web-components#3157]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

spectrum-web-component/picker

### Expected behavior

When user using keyboard, hits enter consecutively 3-4 times, the picker value switches from default to the label.
it should always show the default value.

### Actual behavior

When user using keyboard, hits enter consecutively 3-4 times, the picker value switches from default to the label.

### Screenshots

<img width="739" alt="Screenshot 2023-04-27 at 11 21 19 AM" src="https://user-images.githubusercontent.com/8350272/234771255-c9c5cdde-be06-4c63-b95f-7f3a83732249.png">

### What browsers are you seeing the problem in?

Chrome

### How can we reproduce this issue?

1. Go to 'https://opensource.adobe.com/spectrum-web-components/components/picker/'
2. Click on 'quite picker'
3. select first option from picker, using keyboard , hit enter 3-4 times
4. picker value change from "deselect" to "selection-type"

**Comment by Westbrook** [comment]
After #3456, I am no longer seeing this issue. Feel free to open a new issue with alternative reproduction steps if you continue to have issues with this element.
